docs/manual/forms/variables.rst

```diff
@@ -0,0 +1,170 @@
+auth               object                                Een verzameling van authenticatie gegevens. Zie hieronder.
+auth.plugin        string    ``digid``                   De systeemnaam van de gebruikte authenticatie plugin.
+auth.value         string    ``111222333``               De identificerende waarde in het ``attribute`` van de
+                                                         authenticatie plugin.
+auth_type          string    ``bsn``                     Kan de waarden ``bsn``, ``kvk`` of ``pseudo`` hebben.
+auth_bsn           string    ``111222333``               De waarde van ``auth.value`` indien ``auth_type`` als waarde
+                                                         ``bsn`` heeft. Anders leeg.
+auth_kvk           string    ``90001354``                De waarde van ``auth.value`` indien ``auth_type`` als waarde
+                                                         ``kvk`` heeft. Anders leeg.
+auth_pseudo        string    ``a8bfe7a293dd``...         De waarde van ``auth.value`` indien ``auth_type`` als waarde
+                                                         ``pseudo`` heeft. Anders leeg.
+auth_context       object    ``{"source": "...", ...}``  De volledige authenticatiecontext, met machtigingsinformatie. Zie
+                                                         :ref:`manual_forms_variables_auth_context` voor de beschrijving.
+================== ========= =========================== =========================================================================
+
+**Verouderde variabelen**
+
+Deze variabelen zijn nog wel beschikbaar, maar we raden aan om deze niet meer te
+gebruiken. In versie 3.0 van Open Formulieren kunnen deze verwijderd worden.
+
+=============== ========= =========================== =========================================================================
+Variabele       Type      Voorbeeldwaarde             Toelichting
+=============== ========= =========================== =========================================================================
+auth.machtigen  object                                Verouderd, wordt in 3.0 verwijderd.
+auth.attribute  string    ``bsn``                     Kan de waarden ``bsn``, ``kvk`` of ``pseudo`` hebben (verouderd,
+                                                      gebruik bij voorkeur ``auth_type``).
+=============== ========= =========================== =========================================================================
+
+.. _manual_forms_variables_auth_context:
+
+Authenticatiecontext
+--------------------
+
+De "authenticatie context" bevat alle beschikbare informatie over *hoe* iemand ingelogd
+is op een formulier. Deze bundel informatie is beschikbaar in de vaste variabele
+``auth_context``.
+
+Wanneer er niet ingelogd is op het formulier, dan is de waarde van deze variabele
+``null``.
+
+.. note:: Het is een nieuwere en volledigere variant van de ``auth`` variabele.
+   Er zijn geen plannen om deze laatste te verwijderen.
```

Comment:
Does this mean that `auth.plugin` and `auth.value` (which are both in the static variables table) could also be taken from `auth_context`? Wouldn't that be the preference since it's mentioned that `auth_context` is newer and more complete?

Also it is mentioned that we will not delete `auth`, but above it is mentioned that parts of `auth` (like `machtigen` and maybe `attribute` will be deleted. Perhaps we should put all this in the same info/warning box, because if you just read this info from the note it might seem like using everything from `auth` is fine

Reply:
no, that's also why they aren't removed yet :) IMO there should not be a reason to use `auth.plugin` - the actual plugin used does not matter for semantic checks, instead you want to know if digid/eherkenning auth was used so it's more about the `auth.attribute`. Unfortunately, logic rules are now built on top of `auth.plugin` because people don't know any better.

`auth.value` is a bit more nuanced - it will always contain the identifier of the ultimate beneficiary of the service/product being requested with the form. With `auth_context`, the property holding this value is either `authorizee.legalSubject.identifier` or `representee.identifier`, depending on whether it's digid/eherkenning and/or machtigen is involved or not, which is more complicated to express in form logic.

> because if you just read this info from the note it might seem like using everything from auth is fine

yeah, I'll revisit it tomorrow, the language can definitely be improved!